gems/pending/spec/postgres_ha_admin/failover_databases_spec.rb

```diff
@@ -1,132 +1,93 @@
 require 'postgres_ha_admin/failover_databases'
-require 'postgres_ha_admin/postgres_ha_admin'
-require 'spec/support/test_env_helper'
 require 'pg'
 
 describe PostgresHaAdmin::FailoverDatabases do
-  subject do
-    described_class.new(TestEnvHelper::CONFIG_DIR, TestEnvHelper::CONFIG_DIR, :dbname => 'travis', :user => 'travis')
-    # described_class.new(TestEnvHelper::CONFIG_DIR, TestEnvHelper::CONFIG_DIR, :dbname => 'vmdb_test')
+  before do
+    @yml_file = Tempfile.new('failover_databases.yml')
+    @logger_file = Tempfile.new('ha_admin.log')
+    @logger = Logger.new(@logger_file)
+    @failover_databases = described_class.new(@yml_file, @logger)
```

Comment:
I generally prefer using let over instance variables for most things. Here I would use instance variables for the files as they need to be explicitly closed later, but you should be able to do something like this for the others right at the start of the describe block then use them without the `@`:

```ruby
let(:logger) { Logger.new(@logger_file) }
let(:failover_databases) { described_class.new(@yml_file, logger) }
```